### ISSUE TITLE
Crash on IE11

### ISSUE THREAD

**Issue by oliviertassinari** [apollographql/apollo-client#4024]
**Intended outcome:**

No error.

**Actual outcome:**

`Unable to get property 'get' of undefined or null reference`

![capture d ecran 2018-10-18 a 12 14 12](https://user-images.githubusercontent.com/3165635/47147829-5cb18080-d2cf-11e8-8f13-3241594e59bf.png)

**How to reproduce the issue:**

Go to IE 11 and run:
```js
function getOrCreate(value) {
  var map = new Map();
  return map.get(value) || map.set(value, 'foo').get(value);
}

getOrCreate(undefined)
```

from:
https://github.com/apollographql/apollo-client/blob/51aee1fee8ba7663e64648cd8328e91c0eb52198/packages/apollo-cache-inmemory/src/optimism.ts#L42-L45

**Versions**

```
apollo-cache-inmemory@^1.3.5:
  version "1.3.5"
  resolved "https://registry.yarnpkg.com/apollo-cache-inmemory/-/apollo-cache-inmemory-1.3.5.tgz#7628ee598271cfffa611e15a7ce67652a84b07b0"
  dependencies:
    apollo-cache "^1.1.17"
    apollo-utilities "^1.0.21"
    optimism "^0.6.6"
```

**Comment by oliviertassinari** [comment]
Even better:
`new Map().set('foo', 'bar')` return undefined on IE 11.

**Comment by oliviertassinari** [comment]
The following implementation works:
```js
    CacheKeyNode.prototype.getOrCreate = function (value) {
        var map = this.children || (this.children = new Map);
        var output = map.get(value)
        if (output) {
            return output
        }
        output = new CacheKeyNode()
        map.set(value, output)
        return output
    };
```

**Comment by oliviertassinari** [comment]
https://github.com/zeit/next.js/issues/5426

**Comment by oliviertassinari** [comment]
According to @barnslig, it's a regression, it used to work in v1.2.10.

**Comment by benjamn** [comment]
This should be fixed with `apollo-cache-inmemory@1.3.7` (just published).

**Comment by oliviertassinari** [comment]
Thank you!
